custom/enikshay/integrations/bets/views.py

```diff
@@ -30,67 +31,85 @@ def __init__(self, msg, status_code):
         super(ApiError, self).__init__(msg)
 
 
-class VoucherUpdate(jsonobject.JsonObject):
-    voucher_id = jsonobject.StringProperty(required=True)
-    payment_status = jsonobject.StringProperty(required=True, choices=['success', 'failure'])
-    payment_amount = jsonobject.DecimalProperty(required=False)
-    failure_description = jsonobject.StringProperty(required=False)
+class FlexibleDateTimeProperty(jsonobject.DateTimeProperty):
+    def _wrap(self, value):
+        try:
+            return date_parser.parse(value)
+        except ValueError:
+            return super(FlexibleDateTimeProperty, self)._wrap(value)
 
-    case_type = CASE_TYPE_VOUCHER
+
+class PaymentUpdate(jsonobject.JsonObject):
+    id = jsonobject.StringProperty(required=True)
+    status = jsonobject.StringProperty(required=True, choices=['Success', 'Failure'])
```

Comment:
Could it be worth using constants here? The checks for `self.status == 'Success'` seem fragile.